### PR TITLE
allow copying levels when copying lessons

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes [PLAT-1618](https://codedotorg.atlassian.net/browse/PLAT-1618). 

### before
Screen Shot 2022-03-20 at 7 44 50 PM

### after

Screen Shot 2022-03-18 at 2 38 58 PM

Screen Shot 2022-03-18 at 2 39 12 PM

## Testing story

* Unit test coverage of CloneLessonDialog UI component
* manually verified that deep copy and shallow copy are working properly
* manually verified shallow copy in the case where a level name suffix was previously specified

### CODE REVIEW

File: apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx [hunk 1]
```diff
@@ -4,6 +4,11 @@ import BaseDialog from '@cdo/apps/templates/BaseDialog';
 import Button from '@cdo/apps/templates/Button';
 import DialogFooter from '@cdo/apps/templates/teacherDashboard/DialogFooter';
 
+const LevelCopyText = {
+  deepCopy: 'deep copy: create a new copy of each level in the lesson',
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx]: These dropdown options are very long. Could we use a HelpTip to explain what deep and shallow copy mean in order to keep the dropdown options shorter?

File: apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx [hunk 2]
```diff
@@ -103,6 +116,33 @@ export default class CloneLessonDialog extends Component {
...
+                ))}
+              </select>
+            </label>
+            {this.state.levelCopyType === 'deepCopy' && (
+              <label>
+                Suffix to add when copying levels:{' '}
+                <input
+                  type="text"
+                  value={this.state.newLevelSuffix}
+                  onChange={e =>
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx:158]: If you set this and then pick shallow copy does this value clear?

Comment (dmcavoy) [reply apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx:158]: It looks like you get around this with line 38/39 so I guess it doesn't need to clear.

File: dashboard/app/controllers/lessons_controller.rb [hunk 3]
```diff
@@ -149,8 +149,9 @@ def clone
     raise "Cannot find script #{params[:destinationUnitName]}" unless destination_script
     raise 'Destination script and lesson script must be in a course version' unless destination_script.get_course_version && @lesson.script.get_course_version
     raise 'Lessons current unit and destination unit must both use code studio lesson plans' unless !destination_script.use_legacy_lesson_plans && !@lesson.script.use_legacy_lesson_plans
+    new_level_suffix = params[:newLevelSuffix].presence
```
Comment (dmcavoy) [inline dashboard/app/controllers/lessons_controller.rb:152]: TIL :)

**Review by dmcavoy — APPROVED** [review]
Looks good. Left a couple wonders but not blockers

File: apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx (hunk 2 above)
Comment (davidsbailey) [reply apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx:158]: right, I'm not actually clearing the value of the new level suffix in the react state, then I'm using a unit test to make sure we still send the right data to the server.

File: apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx (hunk 1 above)
Comment (davidsbailey) [reply apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx]: a dropdown in a dialog wasn't working for me. how about this? cc @tess323 

Screen Shot 2022-03-21 at 12 57 47 PM

Comment (tess323) [reply apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx]: Would it work to put the table behind a '?' tooltip?  If not I am not I am totally fine with this.

Comment (davidsbailey) [reply apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx]: a tooltip isn't working for me, due to it being in a dialog, unfortunately.

Comment (tess323) [reply apps/src/lib/levelbuilder/unit-editor/CloneLessonDialog.jsx]: makes total sense - this works for me!

**Pull request merged**